packages/quantic/force-app/main/default/lwc/quanticResultLink/__tests__/quanticResultLink.test.js

```diff
@@ -0,0 +1,114 @@
+// @ts-nocheck
+import '@testing-library/jest-dom';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+import {getNavigateCalledWith} from 'lightning/navigation';
+import {createElement} from 'lwc';
+import QuanticResultLink from '../quanticResultLink';
+import mockDefaultResult from './data/defaultResult.json';
+import mockKnowledgeArticleResult from './data/knowledgeArticleResult.json';
+import mockSalesforceResult from './data/salesforceResult.json';
+
+jest.mock('c/quanticHeadlessLoader');
+let spy = {};
+
+function createTestComponent(options) {
+  const element = createElement('c-quantic-result-link', {
+    is: QuanticResultLink,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+// Helper function to mock headless for this suite of tests.
+function mockHeadless() {
+  const mockHeadlessBundle = {
+    buildInteractiveResult: jest.fn(),
+  };
+
+  jest.mock('../../quanticHeadlessLoader/quanticHeadlessLoader.js', () => ({
+    getHeadlessBundle: jest.fn().mockReturnValue(mockHeadlessBundle),
+  }));
+
+  spy = jest
+    .spyOn(mockHeadlessLoader, 'getHeadlessEnginePromise')
+    .mockReturnValue(
+      new Promise((resolve) => {
+        resolve();
+      })
+    );
+}
+
+describe('c-quantic-result-link', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM.
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  }
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when the result is of type salesforce', () => {
+    it('should open the result link in a salesforce console subTab', async () => {
+      let MOCK_RECORD_ID = '123';
+      mockHeadless();
+
+      const element = createTestComponent({...mockSalesforceResult});
+      await flushPromises();
+
+      const linkSalesforce = element.shadowRoot.querySelector('a');
+      linkSalesforce.click();
+
+      const {pageReference} = getNavigateCalledWith();
+
+      expect(spy).toHaveBeenCalled();
+      expect(pageReference.attributes.recordId).toBe(MOCK_RECORD_ID);
+    });
+
+    describe('is a knowledge article', () => {
+      it('should also open the result link in a salesforce console subTab', async () => {
```

Comment:
```suggestion
      it('should open the result link in a salesforce console subTab', async () => {
```
IMO it's best to treat the tests as independent from each other, even in the naming. Ideally the order in which they're declared in the spec file shouldn't matter.